src/main/java/tech/units/indriya/function/DefaultNumberSystem.java

```diff
@@ -765,23 +765,28 @@ private Number multiplyWideAndNarrow(
         
         if(narrow instanceof Double || narrow instanceof Float) {
             // not converting to BigDecimal, because fractional multiplication is not sensitive to precision loss
-            return wide.doubleValue() * narrow.doubleValue();
+            if (wide instanceof Float && narrow instanceof Float) {
+                // return float if both are floats
+                return wide.floatValue() * narrow.floatValue();
```

Comment:
I'd rather not do that, as by extension to double we can guarantee the float value multiplication not to overflow. Its a corner case granted, but why risk that?